hypershift-operator/controllers/hostedclustersizing/hostedclustersizing_controller.go

```diff
@@ -328,30 +328,35 @@ func (r *reconciler) reconcile(
 		}
 	}
 
-	hostedClusters, err := r.listHostedClusters(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to list hosted clusters when calculating concurrency: %w", err)
-	}
+	// For new clusters being added to the fleet, we have an SLA on creation time and can't afford to delay
+	// the first transition, as it is required for the control plane to schedule. For other clusters, though,
+	// we want to limit the amount of churn happening in order to promote the stability of the management plane.
+	if scheduled := hostedCluster.Annotations[hypershiftv1beta1.HostedClusterScheduledAnnotation]; sizeClassLabelPresent && scheduled == "true" {
```

Comment:
why `sizeClassLabelPresent`? Legacy clusters will be scheduled but not have a size label and we do want the concurrency limit to apply to them.

Reply:
oops I think this should have been an `||` ... or we can just omit `sizeClassLabelPresent`